paymentsheet-example/src/main/java/com/stripe/android/paymentsheet/example/samples/ui/server_side_confirm/ServerSideConfirmationActivity.kt

```diff
@@ -0,0 +1,111 @@
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        flowController = PaymentSheet.FlowController.create(
+            activity = this,
+            paymentOptionCallback = viewModel::handlePaymentOptionChanged,
+            createIntentCallbackForServerSideConfirmation = viewModel::createAndConfirmIntent,
+            paymentResultCallback = viewModel::handlePaymentSheetResult,
+        )
+
+        setContent {
+            MaterialTheme {
+                val uiState by viewModel.state.collectAsState()
+                val paymentMethodLabel = determinePaymentMethodLabel(uiState)
+
+                uiState.status?.let {
+                    LaunchedEffect(it) {
+                        snackbar.setText(it).show()
+                        viewModel.statusDisplayed()
+                    }
+                }
+
+                LaunchedEffect(uiState.requiresFlowControllerConfigure) {
+                    if (uiState.requiresFlowControllerConfigure) {
+                        flowController.configureWithIntentConfiguration(
+                            intentConfiguration = uiState.cartState.toIntentConfiguration(),
+                            configuration = uiState.paymentSheetConfig,
+                            callback = viewModel::handleFlowControllerConfigured,
+                        )
+                    }
+                }
+
+                Receipt(
+                    isLoading = uiState.isProcessing,
+                    cartState = uiState.cartState,
+                    isEditable = true,
+                    onQuantityChanged = viewModel::updateQuantity,
+                ) {
+                    PaymentMethodSelector(
+                        isEnabled = uiState.isPaymentMethodButtonEnabled,
+                        paymentMethodLabel = paymentMethodLabel,
+                        paymentMethodIcon = uiState.paymentOption?.icon(),
+                        onClick = flowController::presentPaymentOptions,
+                    )
+
+                    SubscriptionToggle(
+                        checked = uiState.cartState.isSubscription,
+                        onCheckedChange = viewModel::updateSubscription,
+                    )
+
+                    BuyButton(
+                        buyButtonEnabled = uiState.isBuyButtonEnabled,
+                        onClick = {
+                            viewModel.handleBuyButtonPressed()
+                            flowController.confirm()
+                        }
+                    )
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun determinePaymentMethodLabel(uiState: ServerSideConfirmationViewState): String {
+    val context = LocalContext.current
+    return remember(uiState) {
+        if (uiState.paymentOption?.label != null) {
+            uiState.paymentOption.label
+        } else if (!uiState.isProcessing) {
+            context.getString(R.string.select)
```

Comment:
Could use `stringResource` instead of `context.getString`

Reply:
Not inside the `remember` block.